js/plugins/iq.js

```diff
@@ -0,0 +1,348 @@
+   * @public
+   * @method IqPlugin#setPluginID
+   * @return  {string} The pluginID assigned to this instance from the Analytics Framework.
+   */
+  this.getPluginID = function()
+  {
+    return id;
+  };
+
+  /**
+   * [Required Function] Initialize the plugin with the given metadata.
+   * @public
+   * @method IqPlugin#init
+   */
+  this.init = function()
+  {
+    var missedEvents;
+    //if you need to process missed events, here is an example
+    if (_framework && OO._.isFunction(_framework.getRecordedEvents))
+    {
+      missedEvents = _framework.getRecordedEvents();
+    }
+    //use recorded events.
+
+    if (this.testMode)
+    {
+      trySetupAnalytics();
+    }
+    else if (!window.Ooyala)
+    {
+      OO.loadScriptOnce("https://analytics.ooyala.com/static/v3/analytics.js", trySetupAnalytics, sdkLoadError, SDK_LOAD_TIMEOUT);
+    }
+  };
+
+  /**
+   * [Required Function] Set the metadata for this plugin.
+   * @public
+   * @method IqPlugin#setMetadata
+   * @param  {object} metadata The metadata for this plugin
+   */
+  this.setMetadata = function(metadata)
+  {
+      OO.log( "Analytics Template: PluginID \'" + id + "\' received this metadata:", metadata);
+  };
+
+  /**
+   * [Required Function] Process an event from the Analytics Framework, with the given parameters.
+   * @public
+   * @method IqPlugin#processEvent
+   * @param  {string} eventName Name of the event
+   * @param  {Array} params     Array of parameters sent with the event
+   */
+  this.processEvent = function(eventName, params)
+  {
+    OO.log( "Analytics Template: PluginID \'" + id + "\' received this event \'" + eventName + "\' with these params:", params);
+    switch(eventName)
+    {
+      case OO.Analytics.EVENTS.STREAM_TYPE_UPDATED:
+        if (params && params[0])
+        {
+          //Retrieve the stream type here.
+          //Possible values include OO.Analytics.STREAM_TYPE.VOD and OO.Analytics.STREAM_TYPE.LIVE_STREAM
+          var streamType = params[0].streamType;
+        }
+        break;
+
+      case OO.Analytics.EVENTS.VIDEO_CONTENT_METADATA_UPDATED:
+        if (params && params[0])
+        {
+          duration = params[0].duration;
+          if (this.ooyalaReporter)
+          {
+            this.ooyalaReporter.initializeMedia(currentEmbedCode, contentType);
+            OO.log("IQ: Reported: initializeMedia() with args: " + currentEmbedCode + ", " + contentType);
+            this.ooyalaReporter.setMediaDuration(duration);
+            OO.log("IQ: Reported: setMediaDuration() with args: " + duration);
+          }
+          else
+          {
+            OO.log("Tried reporting event: " + OO.Analytics.EVENTS.VIDEO_CONTENT_METADATA_UPDATED +
+                   " but ooyalaReporter is: " + this.ooyalaReporter);
+          }
+        }
+        break;
+
+      case OO.Analytics.EVENTS.VIDEO_SOURCE_CHANGED:
+        if (params && params[0])
+        {
+          autoPlay = params[0].metadata.autoPlay;
```

Comment:
Need a null check for metadata.

Reply:
@pilievOoyala should the null check for metadata be in the enclosing if statement? Or should it only apply to the assignment of autoPlay?

Reply:
Added a null check in the main if conditional, but please see the above comment